core/templates/pages/learner-dashboard-page/learner-dashboard-page.component.ts

```diff
@@ -346,6 +346,20 @@ angular.module('oppia').component('learnerDashboardPage', {
         );
       };
 
+      /**
+       * This was needed because javascript returns -1 using inBuilt indexOf
+       * even when there are identical objects.
+       * Refer -> https://stackoverflow.com/q/11704971
+       */
+      var findIndex = (arr, element) => {
+        for (var i = 0; i < arr.length; i++) {
+          if (JSON.stringify(arr[i]) === JSON.stringify(element)) {
+            return i;
+          }
+        }
+        return -1;
+      };
+
```

Comment:
Please change this per discussion in comment thread [here](https://github.com/oppia/oppia/pull/9798#discussion_r451844989).

Reply:
done